src/DDTrace/Sampling/ConfigurableSampler.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace DDTrace\Sampling;
+
+use DDTrace\Configuration;
+use DDTrace\Contracts\Span;
+
+/**
+ * A sampler configurable using global a global configuration parameter.
+ */
+class ConfigurableSampler implements Sampler
+{
+    /**
+     * @param Span $span
+     * @return int
+     */
+    public function getPrioritySampling(Span $span)
+    {
+        $rate = Configuration::get()->getSamplingRate();
+
+        if ($rate === 1.0) {
+            return PrioritySampling::AUTO_KEEP;
+        } elseif ($rate === 0.0) {
+            return PrioritySampling::AUTO_REJECT;
+        }
+
+        // NOTE: we do not apply the knuth hashing algorithm here and this is fine for the sole purpose of setting the
+        // sampling priority. This would not work though if we were using this value for client sampling, as the value
+        // and statistical distribution should exactly match the one used by the agent.
+        $shouldKeep = (int) $span->getSpanId() <= $rate * PHP_INT_MAX;
```

Comment:
It just occurred to me that for the trace and span ID generation, we're assuming [`mt_rand()` can spit out 64-bit ints](https://github.com/DataDog/dd-trace-php/blob/5bb3432483280d4e4e1354b0b6987d00f9484dbb/src/DDTrace/ID.php#L17), when in fact [it's maxed out at 32-bit](https://github.com/php/php-src/blob/67e0138c0dfd966624223911a0821f6c294ad1c6/ext/standard/php_mt_rand.h#L30). And that would skew this rate calculation. For now I'm guessing we need to do something like this:
```suggestion
        $shouldKeep = (int) $span->getSpanId() <= $rate * mt_getrandmax();
```
And we should probably update `DDTrace\ID::generate()` to use `mt_getrandmax()` instead of `PHP_INT_MAX` as well. If we wanted 64-bit ints, we could switch to `random_int()`, but it's slower since it's designed for use in crypto. Also it's only PHP 7+ so that might not be viable. Or do you have another suggestion? :)

Reply:
Very good catch! For the current purpose I think that the change you suggested makes sense. as alternative suggestions, looking around seems that combining 2 32 bits random numbers could serve our usage. Let's investigate this in a different PR!